pyomo/repn/plugins/lp_writer.py

```diff
@@ -427,8 +427,6 @@ def write(self, model):
 
             # Pull out the constant: we will move it to the bounds
             offset = repn.constant
-            if offset.__class__ not in int_float:
-                offset = float(offset)
```

Comment:
Why don't you have to cast these anymore? Is it a change from this PR, or was it already redundant?

Reply:
The new walkers are better at catching non-float-like types when processing the leaf nodes, so we no longer have to re-verify the return types from the walker.